pkg/execution/executor/executor.go

```diff
@@ -1893,44 +1948,41 @@ func (e *executor) handleGeneratorInvokeFunction(ctx context.Context, gen state.
 
 	eventName := event.FnFinishedName
 	correlationID := item.Identifier.RunID.String() + "." + gen.ID
-	strExpr := fmt.Sprintf("async.data.%s == %s", consts.InvokeCorrelationId, strconv.Quote(correlationID))
-	_, err = e.newExpressionEvaluator(ctx, strExpr)
-	if err != nil {
-		return execError{err: fmt.Errorf("failed to create expression to wait for invoked function completion: %w", err)}
-	}
 
-	logger.From(ctx).Info().Interface("opts", opts).Time("expires", expires).Str("event", eventName).Str("expr", strExpr).Msg("parsed invoke function opts")
+	logger.From(ctx).Info().Interface("opts", opts).Time("expires", expires).Str("event", eventName).Msg("parsed invoke function opts")
 
 	pauseID := uuid.NewSHA1(
 		uuid.NameSpaceOID,
 		[]byte(item.Identifier.RunID.String()+gen.ID),
 	)
 
 	opcode := gen.Op.String()
+
 	err = e.sm.SavePause(ctx, state.Pause{
-		ID:          pauseID,
-		WorkspaceID: item.WorkspaceID,
-		Identifier:  item.Identifier,
-		GroupID:     item.GroupID,
-		Outgoing:    gen.ID,
-		Incoming:    edge.Edge.Incoming,
-		StepName:    gen.UserDefinedName(),
-		Opcode:      &opcode,
-		Expires:     state.Time(expires),
-		Event:       &eventName,
-		Expression:  &strExpr,
-		DataKey:     gen.ID,
+		ID:                  pauseID,
+		WorkspaceID:         item.WorkspaceID,
+		Identifier:          item.Identifier,
+		GroupID:             item.GroupID,
+		Outgoing:            gen.ID,
+		Incoming:            edge.Edge.Incoming,
+		StepName:            gen.UserDefinedName(),
+		Opcode:              &opcode,
+		Expires:             state.Time(expires),
+		Event:               &eventName,
+		DataKey:             gen.ID,
+		InvokeCorrelationID: &correlationID,
 	})
 	if err == state.ErrPauseAlreadyExists {
 		return nil
 	}
+
 	if err != nil {
+		// TODO: should we delete both keys if failed?
```

Comment:
what do we want to do here?
should we attempt to clean up? otherwise there might be trash data sitting around.

Reply:
Ah, this isn't relevant any more - it's an old TODO from when this had a waitgroup.